### PR TITLE
Add SupplyChain overview document.

### PR DESCRIPTION
**Pull request by cianx**
Signed-off-by: Cian Montgomery <cian.montgomery@intel.com>

### CODE REVIEW

File: docs/source/examples/supplychain/overview.rst [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+Records
++++++++
+
+Records, as we discussed above, represent physical objects in the world.
+Records are identified in the system by `natural keys
+<https://en.wikipedia.org/wiki/Natural_key>`_. These natural keys could be the
+serial number of the object, the RFID tag number, etc.
+
+Records keep a list of all the Owners and Custodians that have possesed the
```
Comment (ojalatodd) [inline docs/source/examples/supplychain/overview.rst]: 1. Separate into separate sentences or use colon 
2. spelling - possesed to possessed
3. typo: change **for** to **from** 

"Records keep a list of all the Owners and Custodians that have  **possessed** the item; these records are stored in chronological order **from** oldest to most recent."

File: docs/source/examples/supplychain/overview.rst [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+.. literalinclude:: ../../../../families/supplychain/protos/record.proto
+    :language: protobuf
+    :lines: 22-
+
+
+Applications
+++++++++++++
+
+Applications are the mechanism that Agents change roles(Owner or Custodian).
```
Comment (ojalatodd) [inline docs/source/examples/supplychain/overview.rst]: Rewrite for sense. Suggestion:
Applications are the mechanism that Agents **use to** change roles (Owner or Custodian).

File: docs/source/examples/supplychain/overview.rst [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+Agents that wish to assume a new role create a  new application for the Record.
```
Comment (ojalatodd) [inline docs/source/examples/supplychain/overview.rst]: Combine these two sentences into one sentence, e.g.:

"Agents that wish to assume a new role for a Record create a new application that specifies the desired role."

File: docs/source/examples/supplychain/overview.rst [hunk 4]
```diff
@@ -0,0 +1,100 @@
+*******************
+SupplyChain Example
+*******************
+
+In this example provided the infrastructure necessary track physical objects
+thru a real world supply chain on the Sawtooth Ledger. Obviously we can not put
```
Comment (ojalatodd) [inline docs/source/examples/supplychain/overview.rst]: Reword similar to the Overview section of the other document ( sawtooth-core/docs/source/examples/supplychain/supplychain_transaction_family.rst )
for sense and flow.

Comment (ojalatodd) [inline docs/source/examples/supplychain/overview.rst]: Question: why is there an **Overview** document and a main document?
Also, regarding the use of "their" or "they" as a gender-neutral singular pronoun, it turns out I was wrong. It recently became accepted to use "they/their" as a gender-neutral, singular pronoun. Such is the evolution of language. ;)

Comment (cianx) [inline docs/source/examples/supplychain/overview.rst]: @ojalatodd I expect other examples to be added to the docs/source/examples.rst document over time.  Such as the "private utxo" documents I am working on now In the future xo, intkey, etc can all have dedicated section where their specs and other parts of the implementation are described.

Comment (ojalatodd) [inline docs/source/examples/supplychain/overview.rst]: @cianx  Great. Approved.

File: docs/source/examples/supplychain/overview.rst [hunk 5]
```diff
@@ -0,0 +1,100 @@
+*******************
+SupplyChain Example
+*******************
+
+In this example provided the infrastructure necessary track physical objects
```
Comment (jsmitchell) [inline docs/source/examples/supplychain/overview.rst]: Suggest rewording to "This example tracks physical objects on Sawtooth using the Supplychain Transaction Family."

File: docs/source/examples/supplychain/overview.rst [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+thru a real world supply chain on the Sawtooth Ledger. Obviously we can not put
+physical objects in to in to the Digital Ledger. We track Records on the Ledger
+which are uniquely linked to the physical object they represent.
+
+This example provides the infrastructure necessary to track physical objects in
```
Comment (jsmitchell) [inline docs/source/examples/supplychain/overview.rst]: This paragraph reads as a duplicate of the above, but seems cleaner. Suggest just deleting the above paragraph and using this one.

File: docs/source/examples/supplychain/overview.rst [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+
+
+SupplyChain Overview
+=====================
+
+Agents
+++++++
+
+Agents represent actors in the system. An Agent is expected most commonly
+to represent a company. Agents are identified by their public keys, any
```
Comment (jsmitchell) [inline docs/source/examples/supplychain/overview.rst]: s/public keys, any transactions/public keys and any transactions/

File: docs/source/examples/supplychain/overview.rst [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+transactions an agent submits must be signed by the agent. Agents serve two
+roles in the system: owners and custodians. Owners are Agents that are the
+legal holders of the physical object a record represents. Custodians are Agents
+that have possesion of the Physical object that a Record represents. Custodians
```
Comment (jsmitchell) [inline docs/source/examples/supplychain/overview.rst]: s/Physical/physical/

File: docs/source/examples/supplychain/overview.rst [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+The ledger representation of "Agents" is:
+
+.. literalinclude:: ../../../../families/supplychain/protos/agent.proto
+    :language: protobuf
+    :lines: 22-
+
+Records
++++++++
+
+Records, as we discussed above, represent physical objects in the world.
```
Comment (jsmitchell) [inline docs/source/examples/supplychain/overview.rst]: s/, as we discussed above,//

File: docs/source/examples/supplychain/overview.rst [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+.. literalinclude:: ../../../../families/supplychain/protos/record.proto
+    :language: protobuf
+    :lines: 22-
+
+
+Applications
+++++++++++++
+
+Applications are the mechanism that Agents use to change roles (Owner or
+Custodian). Agents that wish to assume a new role for a Record create a new application that specifies the desired role. The current Agent in that role
```
Comment (jsmitchell) [inline docs/source/examples/supplychain/overview.rst]: 80 column wrap

File: docs/source/examples/supplychain/overview.rst [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+must then accept the Application before the  Record is updated. This allows
+both Agents to assert their agreement to the change  before the update is made.
```
Comment (jsmitchell) [inline docs/source/examples/supplychain/overview.rst]: there is a double space between "change" and "before"

File: docs/source/examples/supplychain/overview.rst [hunk 12]
```diff
@@ -0,0 +1,100 @@
...
+The applicant(proposing Agent) may cancel the Application or the current Agent
+my reject the Application if it is not wanted.
```
Comment (jsmitchell) [inline docs/source/examples/supplychain/overview.rst]: s/my/may/
